mmv1/third_party/terraform/utils/framework_location_test.go

```diff
@@ -0,0 +1,251 @@
+package google
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func TestLocationDescription_getZone(t *testing.T) {
+	cases := map[string]struct {
+		ld            LocationDescription
+		ExpectedZone  types.String
+		ExpectedError bool
+	}{
+		"returns the value of the zone field in resource config": {
+			ld: LocationDescription{
+				// A resource would not have all 3 fields set, but if they were all present zone is used
+				ResourceZone:     types.StringValue("resource-zone"),
+				ResourceRegion:   types.StringValue("resource-region"),
+				ResourceLocation: types.StringValue("resource-location"),
+				// Provider config doesn't override resource config
+				ProviderRegion: types.StringValue("provider-region"),
+				ProviderZone:   types.StringValue("provider-zone"),
+			},
+			ExpectedZone: types.StringValue("resource-zone"),
+		},
+		"shortens zone values set as self links in the resource config": {
+			ld: LocationDescription{
+				ResourceZone: types.StringValue("https://www.googleapis.com/compute/v1/projects/my-project/zones/us-central1-a"),
+			},
+			ExpectedZone: types.StringValue("us-central1-a"),
+		},
+		"returns the value of the zone field in provider config when zone is unset in resource config": {
+			ld: LocationDescription{
+				ProviderZone: types.StringValue("provider-zone"),
+			},
+			ExpectedZone: types.StringValue("provider-zone"),
+		},
+		"returns an error when a zone value can't be found": {
```

Comment:
could we add location / region values to this test (or as separate tests) to make sure they're ignored?

Reply:
Will do!